### PR TITLE
Disabled context menu items for unsaved files

### LINKED ISSUE #11763: Dilato: Core: The new created file under the Working Files can't be renamed/showed in File Tree/Showed in Explorer with no message, if it isn't saved.
Steps:
1. Install and launch the Brackets
2. Click File menu and click New, then user would create a file under the Working Files
3. Right click on the file, click Rename/Show in File Tree/Show in Explorer

Result: There is no response after click these, if the file is not saved
Expect: There is an error message pop up to prompt user to save the file
noresponse

### PR DESCRIPTION
**Pull request by mansimarkaur**
Fixes #11763

### CODE REVIEW

**Comment by ficristo** [comment]
Have you checked what other editors do in these cases?
In general you need some jsdocs.
@petetnt @zaggino could someone of you review?

File: src/command/DefaultMenus.js [hunk 1]
```diff
@@ -30,7 +30,27 @@ define(function (require, exports, module) {
     var AppInit         = require("utils/AppInit"),
         Commands        = require("command/Commands"),
         Menus           = require("command/Menus"),
-        Strings         = require("strings");
+        Strings         = require("strings"),
+        MainViewManager = require("view/MainViewManager"),
+        CommandManager  = require("command/CommandManager");    
+
+    function _setContextMenuItemsVisible(enabled, items) {
```
Comment (petetnt) [inline src/command/DefaultMenus.js]: Needs JSDoc

File: src/command/DefaultMenus.js [hunk 2]
```diff
@@ -30,7 +30,27 @@ define(function (require, exports, module) {
...
+        items.forEach(function (item) {
+            CommandManager.get(item).setEnabled(enabled);
+        });
+    }
+    
+    function _setMenuItemsVisible() {
```
Comment (petetnt) [inline src/command/DefaultMenus.js]: Needs JSDoc

**Review by petetnt — CHANGES_REQUESTED** [review]
LGTM sans couple of missing pieces of JSDoc

**Comment by petetnt** [comment]
Thanks for this great PR @mansimarkaur! Tested against latest master and it works great.

Could you add few small pieces of JSDoc for the new methods and maybe simple unit test for the functionality? (`it should disable context menu items when not present` and `it should not disable context menu items when present`)

👍

**Comment by mansimarkaur** [comment]
@petetnt  sure, I'll add the required changes.

**Comment by mansimarkaur** [comment]
@petetnt Could you please help me with the tests? I was thinking of adding tests in the `test/Menu-test.js` file. Please let me know if it's right?

**Comment by petetnt** [comment]
@mansimarkaur `test/Menu-test.js` sounds like a good place.

Any specific things you need help with the tests? The structure should could be something like:

```
it("it should disable context menu items when file doesn't exist ", function () {
  // runs create a new file
  // opens context menu
  // checks that all the items are disabled
})
```
```
it("it should disable context menu items when file doesn't exist ", function () {
  // runs create a new file
  // opens context menu
  // checks that all the items are disabled
})
```

```
it("it should enable context menu items when file does exist ", function () {
  // runs create a new file and saves it
  // opens context menu
  // checks that all the items are enabled
})
```

For example check `DocumentCommandHandlers` tests to see how the working set can be manipulated.

**Comment by Denisov21** [comment]
+1 big idea!

**Comment by ficristo** [comment]
This needs more work. Closing for now.

**Review by petetnt — APPROVED** [review]

**Comment by petetnt** [comment]
LGTM sans tests, merging this and will open another issue for the tests as per https://github.com/adobe/brackets/issues/11763#issuecomment-283425634

Thanks for this great contribution @mansimarkaur and sorry for taking so long! 

PS: did this test totally flake? https://travis-ci.org/adobe/brackets/builds/206719157 CLA checked manually just in case.

**Pull request merged**